File: wire-moshi-adapter/src/main/java/com/squareup/wire/Uint64.java
package com.squareup.wire;

import com.squareup.moshi.JsonQualifier;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Annotates longs that should be encoded as unsigned. For example, -1L is encoded as
 * 18446744073709551615.
 */
@Retention(RUNTIME)
@JsonQualifier
@interface Uint64 {
}

File: wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/Uint64.kt
package com.squareup.wire

import com.squareup.moshi.JsonQualifier

/**
 * Annotates longs that should be encoded as unsigned. For example, -1L is encoded as
 * 18446744073709551615.
 */
@Retention(AnnotationRetention.RUNTIME)
@JsonQualifier
internal annotation class Uint64
